Github action 'checkstatus.yml' fails while status pending

When the status checks are still pending, the github action fails and causes you to get emailed...

https://github.com/SymbiFlow/symbiflow-arch-defs/runs/1337290559?check_suite_focus=true
```
  curl -s --url https://api.github.com/repos/SymbiFlow/symbiflow-arch-defs/commits/b31e4a4336d37ea0f2f48b485eeca697da48c9e2/status \
  --header 'authorization: Bearer ***' | head -n 2 | tail -n 1 | grep "success"
  shell: /bin/bash -e ***0***
Error: Process completed with exit code 1.
```

Maybe it should be something like;
```bash
if curl -s --url https://api.github.com/repos/SymbiFlow/symbiflow-arch-defs/commits/b31e4a4336d37ea0f2f48b485eeca697da48c9e2/status --header 'authorization: Bearer ***' | head -n 2 | tail -n 1 | grep "success"; then
  mkdir install && symbiflow_get_latest_artifact_url > install/latest
fi
```

--- a/.github/workflows/checkstatus.yml
+++ b/.github/workflows/checkstatus.yml
@@ -4,26 +4,41 @@ on:
   status:
 
 jobs:
+
   Check-statuses:
     runs-on: ubuntu-latest
-
     steps:
+
     - name: Check status
-      run: |
-        curl -s --url https://api.github.com/repos/${{ github.repository }}/commits/${{ github.event.sha }}/status \
-        --header 'authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' | head -n 2 | tail -n 1 | grep "success"
+      id: check
+      run: >-
+        curl -fsSL
+        -H 'Accept: application/vnd.github.v3+json'
+        -H 'authorization: Bearer ${{ github.token }}'
+        https://api.github.com/repos/${{ github.repository }}/commits/${{ github.event.sha }}/status
+        | head -n 2
+        | tail -n 1
+        | grep "success"
+        || echo '::set-output name=skip::true'
+
     - name: Set up Python
+      if: steps.check.outputs.skip != 'true'
       uses: actions/setup-python@v2
       with:
         python-version: 3.8
+
     - name: Install STDM
+      if: steps.check.outputs.skip != 'true'
       run: pip install git+https://github.com/SymbiFlow/symbiflow-tools-data-manager#egg=stdm
+
     - name: Get latest artifact name
+      if: steps.check.outputs.skip != 'true'
       run: mkdir install && symbiflow_get_latest_artifact_url > install/latest
+
     - name: Upload artifact URL file
+      if: steps.check.outputs.skip != 'true'
       uses: weslenng/gcp-storage-sync@master
       env:
-        GCP_SERVICE_ACCOUNT_KEY_FILE: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY_FILE  }}
-        GCP_STORAGE_BUCKET: ${{ secrets.GCP_STORAGE_BUCKET  }}
+        GCP_SERVICE_ACCOUNT_KEY_FILE: ${{ secrets.GCP_SERVICE_ACCOUNT_KEY_FILE }}
+        GCP_STORAGE_BUCKET: ${{ secrets.GCP_STORAGE_BUCKET }}
         SOURCE_DIR: "install"
-

